extensions/amp-intaker-chat-widget/0.1/widget.js

```diff
@@ -0,0 +1,570 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const Intaker = {};
+/**
+ *
+ * @constructor
+ */
+Intaker.Widget = function() {
+  let CookiesAPI = window['CookiesApi'];
+  let chatIsActive = false;
+  let previewChatIsActive = false;
+  const frameId = 'chatter-bot-iframe';
+  const frameContainerId = 'chatter-bot-frame-container';
+  const btnId = 'chatter-bot-launcher';
+  const closeBtnId = 'chatter-bot-widget-close';
+  const restartBtnId = 'chatter-bot-launcher-restart';
+  const INTAKER_CHAT_URL = btoa('INTAKER_CHAT_URL');
+  const INTKER_CHAT_URL = btoa('INTKER_CHAT_URL'); //support for old generated widgets
+  let chatUrlHash = ''; //window[INTAKER_CHAT_URL] || window[INTKER_CHAT_URL];
+  let url = ''; //atob(chatUrlHash);
+  let useQA = window['USE_INTAKER_QA'];
+  let frameContainer, closeBtn, restartBtn, lunchBtn, frame;
+  let api = '';
+  const cssUrl =
+    (useQA
+      ? 'https://intakerclientqa.azurewebsites.net'
+      : 'https://intaker.co') + '/dist/chat-widget.min.css';
+  let directLink = '';
+  const isDesktop = !isMobile();
+  let cookieName = 'INTAKER_CHAT_WIDGET_';
+  const cookieValue = {};
+  const DEFAULT_AVATAR =
+    'https://intaker.blob.core.windows.net/dashboard/a6.png';
+  let dingSound = null;
+  let launcherContainer = null;
+  let isUniqueVisit = false;
+  let externalUrl = '';
+  const originalTitle = document.title || '';
+  let isAMP = false;
+  let platform = null;
+  let Templates = window['IntakerWidgetTemplates'];
+  let chatUniqueId = null;
+  let manualOpenedChat = false;
+
+  /**
+   *
+   * @param {Element} el
+   * @param {string} name
+   * @return {Element}
+   */
+  function addClass(el, name) {
+    el.classList.add(name);
+    return el;
+  }
+
+  /**
+   *
+   * @param {Element} el
+   * @param {string} name
+   * @return {Element}
+   */
+  function removeClass(el, name) {
+    el.classList.remove(name);
+    return el;
+  }
+
+  /**
+   *
+   * @param {string} url
+   */
+  function loadCss(url) {
+    const link = document.createElement('link');
+    // link.id    = 'chatter-bot-css';
+    link.rel = 'stylesheet';
+    link.type = 'text/css';
+    link.href = url;
+    link.media = 'all';
+    (document.getElementsByTagName('head')[0] || document.body).appendChild(
+      link
+    );
+  }
+
+  /**
+   *
+   * @param {string} string
+   */
+  function addToBody(string) {
+    let tmp = document.createElement('div');
+    tmp./*OK*/ innerHTML = string;
+    document.body.appendChild(tmp.firstChild);
+    tmp = null;
+  }
+
+  /**
+   *
+   */
+  function closeFrame() {
+    exitPreview();
+    visitor(false);
+    chatIsActive = false;
+    removeClass(frameContainer, 'chatter-bot-frame-container-active');
```

Comment:
AMP has utilities for this.

Reply:
where are they? `dom.js` itself is using the pure `classList.remove` function